src/geometry/LineUtil.js

```diff
@@ -240,3 +241,59 @@ export function _flat(latlngs) {
 	console.warn('Deprecated use of _flat, please use L.LineUtil.isFlat instead.');
 	return isFlat(latlngs);
 }
+
+/* @function polylineCenter(latlngs: LatLng[], map: Map): LatLng
+ * Returns the center of the `Polyline` of the passed LatLngs.
+ * The returned LatLng will be on the `Polyline`.
+ */
+export function polylineCenter(latlngs, map) {
+	var i, halfDist, segDist, dist, p1, p2, ratio, center;
+
+	if (!latlngs || latlngs.length === 0) {
+		throw new Error('latlngs not passed');
+	}
+
+	if (!isFlat(latlngs)) {
+		throw new Error('latlngs are not flat!');
+	}
+
+	// throws error when map is not passed because calculation requires projected coordinates
+	if (!map) {
+		throw new Error('map not passed');
+	}
+
+	var zoom = map.getMaxZoom() === Infinity ? map.getZoom() : map.getMaxZoom();

```

Comment:
1. Excessive call of `map.getMaxZoom()`
2. Do you have tests with `Infinity` zoom?
3. Do you have tests with `undefined` zoom?

Reply:
1. replaced - reading out once and then check in the if
2. The most tests are with maxZoom Infinity, or do you mean the actual zoom `getZoom()`? When does this happen?
3. When does this happen?